### ISSUE TITLE
Undergraduate Machine Learning 29

### ISSUE THREAD

**Issue by Yajieniu** [Yajieniu/CMS-Progress-Mark#13]
## Neural nets and backpropagation

1. If you have many data points, i.e. from 0 to n, then the overall likelihood is the product of the likelihood of each data point, i.e. the product of Bernoulli. 
2. Cost is the negative of log likelihood. Maximizing likelihood == minimizing the cross-entropy. 
3. Classify data into many classes -- using one of k encoding. k is the number of classes. Softmax function is defined to get a probabilistic model. The likelihood for a single data point looks like the Bernoulli but for three classes.
4. Backpropagation (the following steps are not accurate)
     1) Get the cost functions, or equivalently the maximum likelihood
     2) Regularize them by adding theta transpose theta (....?)
     3) Get an estimate of theta -- differentiate the log likelihood and equate it to 0
     4) Do gradient descent
     5) Do batches: derivations with respect to each theta.
     6) Do online
